test/src/Tool/ProviderRedirectTraitTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace League\OAuth2\Client\Test\Tool;
+
+use Eloquent\Phony\Phpunit\Phony;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\BadResponseException;
+use League\OAuth2\Client\Tool\ProviderRedirectTrait;
+use PHPUnit_Framework_TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\RequestInterface;
+
+class ProviderRedirectTraitTest extends PHPUnit_Framework_TestCase
+{
+    use ProviderRedirectTrait;
+
+    public function getHttpClient()
+    {
+        return $this->httpClient;
+    }
+
+    public function setHttpClient(ClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+
+        return $this;
+    }
+
+    public function testRedirectLimitDefault()
+    {
+        $this->assertEquals(2, $this->getRedirectLimit());
+    }
+
+    public function testSetRedirectLimit()
+    {
+        $redirectLimit = rand(3, 5);
+        $this->setRedirectLimit($redirectLimit);
+        $this->assertEquals($redirectLimit, $this->getRedirectLimit());
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     **/
+    public function testSetRedirectLimitThrowsExceptionWhenNonNumericProvided()
+    {
+        $redirectLimit = 'florp';
+        $this->setRedirectLimit($redirectLimit);
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     **/
+    public function testSetRedirectLimitThrowsExceptionWhenZeroProvided()
+    {
+        $redirectLimit = 0;
+        $this->setRedirectLimit($redirectLimit);
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
```

Comment:
Would be nice if these used the methods to expect exceptions instead of docblocks. Not something worth holding up the PR over.